### ISSUE TITLE
Build emulation does not work on non standard Dockerfile names

### ISSUE THREAD

**Issue by mwohlert** [balena-io/balena-cli#1624]
- **balena CLI version:** 11.25.18
- **Operating system version:** Arch Linux
- **32/64 bit OS and processor:** 64bit
- **Install method:** npm
- **If npm install, Node.js and npm version:** Node v12.6.0 and npm v6.9.0
---

When using a Dockerfile with a diffrent name qemu detection does not work and emulator will not be used.

Example:
Dockerfile
```
FROM balenalib/armv7hf-debian:latest
RUN echo 'hello world'
```

Command:
`balena build --application testApp --source . --logs --emulated --dockerfile Dockerfile.debug`

This fails on x86 with:
```
[Info]    Building for testApp/q-svr-001
[Build]   Building services...
[Build]   main Preparing...
[Info]    Emulation is enabled
[Build]   main Step 1/2 : FROM balenalib/armv7hf-debian:latest
[Build]   main  ---> 98236338c3e5
[Build]   main Step 2/2 : RUN echo 'hello world'
[Build]   main  ---> Running in d798ee31cabe
[Build]   main standard_init_linux.go:211: exec user process caused "exec format error"
[Build]   Built 1 service in 0:20
[Error]   Build failed
The command '/bin/sh -c echo 'hello world'' returned a non-zero code: 1

Additional information may be available by setting a DEBUG=1 environment
variable: "set DEBUG=1" on a Windows command prompt, "$env:DEBUG = 1" on
powershell, or "export DEBUG=1" on Linux or macOS.

If you need help, don't hesitate in contacting our support forums at
https://forums.balena.io

For CLI bug reports or feature requests, have a look at the GitHub issues or
create a new one at: https://github.com/balena-io/balena-cli/issues/
```

However if the Dockerfile is actually named Dockerfile the output is the following:
```
[Info]    Building for testApp/q-svr-001
[Build]   Building services...
[Build]   main Preparing...
[Info]    Emulation is enabled
[Build]   main Step 1/3 : FROM balenalib/armv7hf-debian:latest
[Build]   main  ---> 98236338c3e5
[Build]   main Step 2/3 : COPY [".balena/qemu-execve","/tmp/qemu-execve"]
[Build]   main  ---> 1f5a579b2cd1
[Build]   main Step 3/3 : RUN ["/tmp/qemu-execve","-execve","/bin/sh","-c","echo 'hello world'"]
[Build]   main  ---> Running in e2bbfa4c2d0b
[Build]   main hello world
[Build]   main Removing intermediate container e2bbfa4c2d0b
[Build]   main  ---> f9afc78be00f
[Build]   main Successfully built f9afc78be00f
[Build]   main Successfully tagged qbackend_main:latest
[Build]   main Waiting...
[Build]   Built 1 service in 0:17
[Success] Build succeeded!
```

You can clearly see qemu beeing copied in step 2.

**Comment by pdcastro** [comment]
@mwohlert, thanks for reporting this issue. A couple of PRs are under review with a fix (#1628 and balena-io-modules/docker-qemu-transpose/pull/28).

**Comment by pdcastro** [comment]
This issue should be fixed in CLI release v11.28.4. Thanks again for reporting it.

**Comment by pdcastro** [comment]
@mwohlert, by the way, just a note. I believe the command line you shared in this issue for `balena build` wasn't correct in the way it specified the source folder: the `--source` option is valid for the `push` and `deploy` commands, but not for the `build` command. The following line should work:
```
balena build . --application testApp --logs --emulated --dockerfile Dockerfile.debug
```
